ledger-core/instrumentation/insapp/server_multi.go

```diff
@@ -6,51 +6,57 @@
 package insapp
 
 import (
-	"context"
+	"sync"
 
-	"github.com/insolar/component-manager"
-
-	"github.com/insolar/assured-ledger/ledger-core/appctl/beat"
 	"github.com/insolar/assured-ledger/ledger-core/configuration"
+	"github.com/insolar/assured-ledger/ledger-core/instrumentation/insapp/internal/cloud"
 	"github.com/insolar/assured-ledger/ledger-core/network"
-	"github.com/insolar/assured-ledger/ledger-core/network/messagesender"
-	"github.com/insolar/assured-ledger/ledger-core/network/nodeinfo"
 	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+	"github.com/insolar/component-manager"
 )
 
-// NetworkSupport provides network-related functions to an app compartment
-type NetworkSupport interface {
-	beat.NodeNetwork
-	nodeinfo.CertificateGetter
-
-	CreateMessagesRouter(context.Context) messagesender.MessageRouter
-
-	AddDispatcher(beat.Dispatcher)
-	GetBeatHistory() beat.History
-}
-
 // NetworkInitFunc should instantiate a network support for app compartment by the given configuration and root component manager.
 // Returned NetworkSupport will be registered as a component and used to run app compartment.
 // Returned network.Status will not be registered as a component, it can be nil, then monitoring/admin APIs will not be started.
-type NetworkInitFunc = func(configuration.Configuration, *component.Manager) (NetworkSupport, network.Status, error)
+type NetworkInitFunc = func(configuration.Configuration, *component.Manager) (cloud.NetworkSupport, network.Status, error)
 
 // MultiNodeConfigFunc provides support for multi-node process initialization.
-// For the given config path and base config this handler should return a list of configurations (one per node). And NetworkInitFunc
-// to initialize instantiate a network support for each app compartment (one per node). A default implementation is applied when NetworkInitFunc is nil.
-type MultiNodeConfigFunc = func(cfgPath string, baseCfg configuration.Configuration) ([]configuration.Configuration, NetworkInitFunc)
+// For the given base config this handler should return a list of configurations (one per node).
+type MultiNodeConfigFunc = func(baseCfg configuration.Configuration) []configuration.Configuration
 
-func NewMulti(cfgPath string, appFn AppFactoryFunc, multiFn MultiNodeConfigFunc) *Server {
-	switch {
-	case appFn == nil:
-		panic(throw.IllegalValue())
-	case multiFn == nil:
+func NewMulti(cfg configuration.BaseCloudConfig, multiFn MultiNodeConfigFunc) *MultiServer {
+	if multiFn == nil {
 		panic(throw.IllegalValue())
 	}
 
-	return &Server{
-		cfgPath: cfgPath,
-		appFn:   appFn,
+	return &MultiServer{
+		cfg:     cfg,
 		multiFn: multiFn,
 	}
 }
 
+type MultiServer struct {
+	cfg     configuration.BaseCloudConfig
+	multiFn MultiNodeConfigFunc
+}
+
+func (s *MultiServer) Serve() {
```

Comment:
imho - wrong separation of duties
Serve() should should operations, but should NOT create new instances

Reply:
reverted to common Server